java/core/src/java/org/apache/orc/InMemoryKeystore.java

```diff
@@ -278,16 +276,7 @@ public InMemoryKeystore addKey(String keyName, int version,
       algorithm = EncryptionAlgorithm.AES_CTR_128;
     }
 
-    final byte[] buffer = new byte[algorithm.keyLength()];
-    if (algorithm.keyLength() > masterKey.length) {
-
-      System.arraycopy(masterKey, 0, buffer, 0, masterKey.length);
-      /* fill with zeros */
-      Arrays.fill(buffer, masterKey.length, buffer.length - 1, (byte) 0);
-
-    } else {
-      System.arraycopy(masterKey, 0, buffer, 0, algorithm.keyLength());
-    }
```

Comment:
As I understand this code, it's a bit superfluous.  There is no need to "fill with zeros" as the array will, by default in Java, pre-populated with zero value.

And this:

> For any indices that are valid in the copy but not the original, the copy will contain (byte)0. Such indices will exist if and only if the specified length is greater than that of the original array.

https://docs.oracle.com/javase/7/docs/api/java/util/Arrays.html#copyOf(byte[],%20int)